app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
+  def check
+    unless logged_in_as_admin?
```

Comment:
I'm not sure about the model vs controller, so let's poll the audience! @shalott, do you have an opinion?  :D 

My overall feeling, though, is that if we _do_ move it to the model, each action should be a separate method -- one like clean_up_subscriptions, another like set_works_revised_at_date, another for reindex_users_works, another for reindex_users_bookmarks. (Bad names, but you get the idea.)